client/block.c

```diff
@@ -559,6 +640,50 @@ static int add_block_nolock(struct xdag_block *newBlock, xdag_time_t limit)
 		}
 	}
 
+#if CACHE == 1
+	if(g_xdag_extstats.cache_usage >= g_xdag_extstats.cache_size){
+		if (g_xdag_extstats.cache_hitrate<0.94 && g_xdag_extstats.cache_size*2 <= CACHE_MAX_SIZE){
+			if(!g_xdag_extstats.cache_size && CACHE_MAX_SIZE){
```

Comment:
if(!g_xdag_extstats.cache_size && CACHE_MAX_SIZE){
Is this condition correct? Why do you need to include CACHE_MAX_SIZE in the condition? 
On the first hit g_xdag_extstats.cache_size = 0, so execution goes to g_xdag_extstats.cache_size++ and a new value is 1.
On the next hits cache_size is doubled, so, next values are 2, 4, 8 and etc... Is that logic with cache_size correct?